wolfcrypt/src/sha512.c

```diff
@@ -1425,11 +1463,14 @@ static INLINE void AddLength384(Sha384* sha384, word32 len)
 
 int wc_Sha384Update(Sha384* sha384, const byte* data, word32 len)
 {
+    if (sha384 == NULL || (data == NULL && len > 0)) {
+       return BAD_FUNC_ARG;
+    }
     /* do block size increments */
     byte* local = (byte*)sha384->buffer;
```

Comment:
move variable declarations to beginning of statement or change to "declaration then assignement"